src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditorCore.jsx

```diff
@@ -0,0 +1,239 @@
+  if (!currentRelationships.includes(newRelationship))
+    newRelationships = [...currentRelationships, newRelationship];
+  allRelationships[currentCategory] = newRelationships;
+  return allRelationships;
+}
+
+function renderSingleRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+  onChange,
+) {
+  return (
+    <div style={{ marginLeft: 32, marginTop: 10 }} key={relationship}>
+      <TextInput
+        width={240}
+        schema={relationship}
+        onChange={newRelationshipName => {
+          onChange(
+            updateRelationshipInCategory(
+              allRelationships,
+              currentCategory,
+              relationship,
+              newRelationshipName,
+            ),
+          );
+        }}
+        value={relationship}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationships,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationships,
+  targetKey,
+  onChange,
+) {
+  const relationships = get(allRelationships, targetKey, []);
+  return (
+    <div style={{ marginTop: 10 }} key={targetKey}>
+      <TextInput
+        width={240}
+        schema={targetKey}
+        onChange={newName => {
+          onChange(
+            updateCategoryName(allRelationships, targetKey, newName),
+          );
+        }}
+        value={targetKey}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  onChange(
+                    addRelationshipToCategory(
+                      allRelationships,
+                      targetKey,
+                      '',
+                    ),
+                  );
+                }}
+              >
+                <NewChildIcon />
+              </IconButton>
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteCategory(allRelationships, targetKey),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+      {relationships.map(relationship =>
+        renderSingleRelationship(
+          allRelationships,
+          targetKey,
+          relationship,
+          onChange,
+        ),
+      )}
+    </div>
+  );
+}
+
+export default function RelationshipEditorCore({
+  schema,
+  value,
+  onChange,
+  siteSettings, // unpacking to prevent passing to div
```

Comment:
If you are not going to use this prop, is there a reason to pass it in?

Reply:
> 

The one that live at points older than when I merged in develop I think failed when I tried to rebase them. I'm also worried about re-writing that much of our history at this point.

I'll experiment with this in a backup branch or two tomorrow.